### PR TITLE
fix #2669 conda config --add channels can leave out defaults

### LINKED ISSUE #2669: Having trouble downloading from osx-64 on Travis CI
_From @jakirkham on June 14, 2016 20:46_

Seems we are now having issues downloading on packages on Travis CI. See the snippet below and this [log]( https://travis-ci.org/conda-forge/apache-libcloud-feedstock/jobs/137608834#L369-L376 ). Note none of the packages below are on conda-forge (there is `python`, but not 2.7). So, this seems to be a `defaults` channel issue. Log snippet below.

```
BUILD START: apache-libcloud-0.18.0-py27_1
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata .......
WARNING: Could not find any versions of conda-build in the channels
Solving package specifications ....
Error: Packages missing in current osx-64 channels: 
  - python 2.7*
  - setuptools
```

Also, see these PRs for examples

* https://github.com/conda-forge/apache-libcloud-feedstock/pull/3
* https://github.com/conda-forge/colorlog-feedstock/pull/4
* https://github.com/conda-forge/python-feedstock/pull/20

cc @msarahan @danielfrg @frol @jjhelmus @vamega

_Copied from original issue: Anaconda-Platform/support#45_

### PR DESCRIPTION
**Pull request by kalefranz**

### CODE REVIEW

**Comment by kalefranz** [comment]
@mcg1969 Care to give a quick glance?

**Comment by mcg1969** [comment]
I think we should understand exactly why 4.0.8 is working and 4.1 is not before we commit to anything here. I'm investigating 4.0.8  now

**Comment by kalefranz** [comment]
I'm guessing it has something to do with where I added in the `ruamel_yaml` read.

File: conda/cli/main_config.py [hunk 1]
```diff
@@ -218,26 +218,20 @@ def execute_config(args, parser):
...
+            rc_config = yaml_load(fh)
     else:
-        with open(rc_path, 'r') as rc:
-            rc_text = rc.read()
-    rc_config = yaml_load(rc_text)
-    if rc_config is None:
         rc_config = {}
 
+    # add `defaults` channel if creating new file or channel key doesn't exist in current file
+    if not os.path.exists(rc_path) or 'channels' not in rc_config:
```
Comment (kalefranz) [inline conda/cli/main_config.py]: Actually, this `os.path.exists(rc_path)` could just change to `rc_config`, which will be empty if the path doesn't exist.

Comment (kalefranz) [inline conda/cli/main_config.py]: Actually... I can take the whole first part out.

Comment (mcg1969) [inline conda/cli/main_config.py]: I see no reason for the `not os.path.exists(rc_path) or` part. After all, if `rc_path` doesn't exist, then the dictionary is going to be empty

Comment (kalefranz) [inline conda/cli/main_config.py]: yeah exactly.

**Comment by mcg1969** [comment]
Yeah, I like this implementation better than the old one from 4.0.8

**Comment by kalefranz** [comment]
Ok cleaned it up even a bit more.

**Pull request merged**